x/crosschain/keeper/evm_deposit.go

```diff
@@ -39,21 +43,36 @@ func (k Keeper) HandleEVMDeposit(ctx sdk.Context, cctx *types.CrossChainTx, msg
 		}
 	} else {
 		// cointype is Gas or ERC20; then it could be a ZRC20 deposit/depositAndCall cctx.
-		contract, data, err := parseContractAndData(msg.Message, msg.Asset)
+		parsedAddress, data, err := parseAddressAndData(msg.Message)
 		if err != nil {
-			return false, errors.Wrap(types.ErrUnableToParseContract, err.Error())
+			return false, errors.Wrap(types.ErrUnableToParseAddress, err.Error())
 		}
+		if parsedAddress != (ethcommon.Address{}) {
```

Comment:
Curious why we need allow parsedAddress == ethcommon.Address{} ?

Reply:
This is the fallback when no address is read from the message, it only contains data. So `to` will not be updated and will remains `msg.Receiver`

Currently the fallback is to use `msg.Asset` but it doesn't seem valid to me since `msg.Asset` represents an address on the external chain and not ZetaChain https://github.com/zeta-chain/node/issues/1234